tools/bin/commands/release

```diff
@@ -0,0 +1,149 @@
+
+    --docker-registry         The docker registry
+    --docker-server-id        The docker server id (as defined in settings.xml)
+    --docker-username         The docker username
+    --docker-email            The docker email
+    --docker-password         The docker password
+
+
+EOT
+}
+
+release::run() {
+    source "$(basedir)/commands/build"
+    local dry_run=$(hasflag --dry-run)
+    local release_version=$(readopt --release-version)
+    local dev_version=$(readopt --dev-version)
+    local maven_opts=""
+
+    if [ -z "${release_version}" ]; then
+        echo "Please specify --release-version"
+        exit 1
+    fi
+
+    if [ -z "${dev_version}" ]; then
+        echo "Please specify --dev-version"
+        exit 1
+    fi
+
+    local profiles=$(readopt --profiles)
+    if [ -n "${profiles}" ]; then
+        maven_opts="$maven_opts -P${profiles}"
+    else
+        maven_opts="-Prelease"
+    fi
+
+    local settings_xml=$(readopt --settings-xml)
+    if [ -n "${settings_xml}" ]; then
+        maven_opts="$maven_opts -s $settings_xml"
+    fi
+
+    echo "Creating new staging repository..."
+    local repo_id=$(open_staging_repository "$maven_opts")
+    echo "Opened staging repository: $repo_id"
+    trap "do_cleanup \"$repo_id\" \"none\" \"$maven_opts\"" EXIT
+
+    do_release "$release_version" "$dev_version" "$maven_opts"
+    local tag=$(find_tag)
+    trap "do_cleanup \"$repo_id\" \"none\" \"$maven_opts\"" EXIT
+
+    echo "Closing staging repository: $repo_id"
+    close_staging_repository "$repo_id" "$maven_opts"
+
+    if [ -n "$dry_run" ]; then
+        echo "This is a dry run ..."
+        echo "Droping tag $tag"
+        git tag -d $tag
+        echo "Droping staging repository: $repo_id"
+        drop_staging_repository "$repo_id" "$maven_opts"
+    else
+        echo "Releasing staging repository: $repo_id"
+        trap "echo Done!" EXIT
+        exit
+        release_staging_repository "$repo_id" "$maven_opts"
+        git push origin master
+        git push orign $tag
+    fi
+}
+
+# ======================================================
+# Staging / Release Helper Functions
+
+latest_release() {
+    git log | grep "prepare release $artifactId-" | head -n 1 | awk -F "$artifactId-" '{print \$2}'
+}
+
+open_staging_repository() {
+    local maven_opts=$1
+    local maven_args="org.sonatype.plugins:nexus-staging-maven-plugin:${NEXUS_STAGING_PLUGIN_VERSION}:rc-open -DnexusUrl=${NEXUS_URL} -DserverId=${NEXUS_SERVER_ID} -DstagingProfileId=${NEXUS_DEFAULT_STAGING_PROFILE_ID} -DopenedRepositoryMessageFormat='$REPO_ANNOUNCEMENT_PREFIX:%s' $maven_opts"
+    #Call maven and just grab the last line that contains $REPO_ANNOUNCEMENT_PREFIX discard the line that displays the whole command and get the string that is between `:` and `'`.
+    call_maven "$maven_args" tail -n 1 | grep -v maven | grep $REPO_ANNOUNCEMENT_PREFIX | awk -F ":" '{print $2}' | awk -F "'" '{print $1}'
```

Comment:
Isn't there a `${profile.id}.properties` file and other goals pickup on this, [docs ](https://github.com/sonatype/nexus-maven-plugins/blob/master/staging/maven-plugin/README.md) kinda suggest that is the case, not sure if it holds true

Reply:
My understandings is that this property file is only created by the `deploy` goal and is reused if found by `close`, `release` and `drop`.

Either way, we would still need to do some `cat`, `grep` and `awk` to fech  the staging repository name, so its not something that would change, simplify anything.